pkg/controller/resourceclaim/controller.go

```diff
@@ -152,13 +163,16 @@ func NewController(
 	}
 	if _, err := claimInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
 		AddFunc: func(obj interface{}) {
-			ec.onResourceClaimAddOrUpdate(logger, obj)
+			logger.V(6).Info("new claim", "claimDump", obj)
+			ec.enqueueResourceClaim(logger, obj, false)
 		},
 		UpdateFunc: func(old, updated interface{}) {
-			ec.onResourceClaimAddOrUpdate(logger, updated)
+			logger.V(6).Info("updated claim", "claimDiff", cmp.Diff(old, updated))
```

Comment:
this diff will always run, regardless of whether the result is logged... isn't that expensive? also, cmp.Diff isn't guaranteed to not panic:

From https://pkg.go.dev/github.com/google/go-cmp/cmp:
> Its propensity towards panicking means that its unsuitable for production environments where a spurious panic may be fatal.

Reply:
You are right, let's better guard the call with an if check. I think at -v6 it's okay to take the risk of a panic. I've seen other log calls which dump diffs at higher verbosity levels.

Reply:
> I think at -v6 it's okay to take the risk of a panic

Do we know this is a safe type to pass to cmp.Diff? Should we test that to give us confidence V(6) won't panic?

Reply:
I have tested that manually while developing the PR. For SIG Instrumentation, I am running a periodic job with logging at a very high log level which happens to include DRA. But automated testing in release informing jobs is missing.

I could add a dedicated unit test, but that's not worth it, therefore I have removed the usage of `cmp.Diff`: https://github.com/kubernetes/kubernetes/compare/5cb4f18791a44d23c2d1d83ba7323ec903e1597b..80ab8f0542f9ddcf4935e24f742ef2a94b204471

@liggitt Okay to merge now?

Reply:
> @liggitt Okay to merge now?

yuup